### ISSUE TITLE
Unity output huge amount of warnings on Android.

### ISSUE THREAD

**Issue by plepers** [bugsnag/bugsnag-unity#180]
### Description
Unity output huge amount of warnings on Android.

### Issue
As soon as Bugsnag is initialized, the console is flooded by 2 warnings, triggered by unity, making the app laggy.
The warning are the following:

```
W/Unity: AndroidJNIHelper: converting Byte array is obsolete, use SByte array instead
W/Unity: AndroidJNIHelper.GetSignature: using Byte parameters is obsolete, use SByte parameters
```

### Environment

* bugsnag-unity version: v4.6.7 (android 64bit)
* Unity version: 2019.2.0f1
* Operating system name and version: OSX 10.14.6
* Target platform names and versions: Android - api level 28 - 
* Initializing bugsnag via the Unity UI or in code? UI

### Proposed fix

The issue seem to came from the method `MakeJavaString` in `src/BugsnagUnity/Native/Android/NativeInterface.cs`

https://github.com/bugsnag/bugsnag-unity/blob/f23b033e837f78470ed1de0dcf1212a7df3045ab/src/BugsnagUnity/Native/Android/NativeInterface.cs#L494

`AndroidJavaObject` constructor is call with an utf string encoded as `byte[]`. This type seems to be deprecated and should be replaced by `sbyte[]`.

The following change has fixed the issue for me:

```cs
    private AndroidJavaObject MakeJavaString(string input) {
      if (input == null) {
        return null;
      }

      try {
        var CharsetClass = new AndroidJavaClass("java.nio.charset.Charset");
        // The default encoding on Android is UTF-8
        var charset = CharsetClass.CallStatic<AndroidJavaObject>("defaultCharset");
        byte[] Bytes = Encoding.UTF8.GetBytes(input);
        sbyte[] SBytes = new sbyte[Bytes.Length];
        Buffer.BlockCopy( Bytes, 0, SBytes, 0, Bytes.Length );
        return new AndroidJavaObject("java.lang.String", SBytes, charset);
      } catch (EncoderFallbackException _) {
        // The input string could not be encoded as UTF-8
        return new AndroidJavaObject("java.lang.String");
      }
    }
```

**Comment by mattdyoung** [comment]
Hi @plepers 

Thanks for the report. We'll look into what the best option is for resolving this.

Thanks!

**Comment by JPerryOddGames** [comment]
hi @mattdyoung , is there any ETA for this fix to be implemented? cheers :)

**Comment by xander-jones** [comment]
Hi @JPerryOddGames, we don't have an ETA for this just yet as we have some other prioritized items taking precedence. We hope to have a resolution for this soon!

**Comment by mattdyoung** [comment]
Fixed in [v4.8.2](https://github.com/bugsnag/bugsnag-unity/releases/tag/v4.8.2).
